modules/dns_common/dns_common.rb

```diff
@@ -20,6 +20,77 @@ def resolver
       Resolv::DNS.new(:nameserver => @server)
     end
 
+    def create_a_record(fqdn, ip)
+      case a_record_conflicts(fqdn, ip) #returns -1, 0, 1
+        when 1 then
+          raise(Proxy::Dns::Collision, "'#{fqdn} 'is already in use")
+        when 0 then
+          return nil
+        else
+          do_create(fqdn, ip, "A")
+      end
+    end
+
+    def create_aaaa_record(fqdn, ip)
+      case aaaa_record_conflicts(fqdn, ip) #returns -1, 0, 1
+        when 1 then
+          raise(Proxy::Dns::Collision, "'#{fqdn} 'is already in use")
+        when 0 then
+          return nil
+        else
+          do_create(fqdn, ip, "AAAA")
+      end
+    end
+
+    def create_cname_record(fqdn, target)
+      case cname_record_conflicts(fqdn, target) #returns -1, 0, 1
+        when 1 then
+          raise(Proxy::Dns::Collision, "'#{fqdn} 'is already in use")
+        when 0 then
+          return nil
+        else
+          do_create(fqdn, target, "CNAME")
+      end
+    end
+
+    def create_ptr_record(fqdn, ptr)
+      case ptr_record_conflicts(fqdn, ptr) #returns -1, 0, 1
+        when 1 then
+          raise(Proxy::Dns::Collision, "'#{ptr}' is already in use")
+        when 0
+          return nil
+        else
+          do_create(ptr, fqdn, "PTR")
+      end
+    end
+
+    def do_create(name, value, type)
+      raise(Proxy::Dns::Error, "Creation of #{type} not implemented")
```

Comment:
My thinking was that existing providers will have some create functions, but maybe not all. This would maintain compatibility with existing plugins. Maybe we could deprecate overriding the functions and start requiring this in the 1.16 window.